pandas/core/ops.py

```diff
@@ -663,6 +663,10 @@ def wrapper(left, right, name=name, na_op=na_op):
                     not isinstance(lvalues, pd.DatetimeIndex)):
                 lvalues = lvalues.values
 
+            if (hasattr(rvalues, 'values') and
+                    not isinstance(rvalues, pd.DatetimeIndex)):
```

Comment:
I'm a bit confused by `_TimeOp._convert_to_array()`. It says it converts values to ndarray but it doesn't - at least for some inputs it returns Series. I don't fully understand what it should do so I haven't yet tried to fix or clean it (neither this nor lines 661-664 from ops.py: `if (hasattr(lvalues, 'values')`. As for now, I only modified `to_timedelta` param `box`, which affects `Int64Index`. Maybe the cleaning of `_TimeOp` and co should go to another commit?